homeassistant/components/mqtt/config_flow.py

```diff
@@ -495,6 +514,13 @@ async def async_step_broker(
             step_id="broker", data_schema=vol.Schema(fields), errors=errors
         )
 
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a reconfiguration flow initialized by the user."""
+        self._existing_entry = self._get_reconfigure_entry()
+        return await self.async_step_broker()
```

Comment:
The policy is to avoid caching it, and instead get it fresh inside the broker step, together with `if self.source == SOURCE_RECONFIGURE`

```suggestion
        return await self.async_step_broker()
```